lib/dns.js

```diff
@@ -87,11 +90,13 @@ function onlookup(err, addresses) {
 
 
 function onlookupall(err, addresses) {
-  var results = [];
   if (err) {
-    return this.callback(errnoException(err, 'getaddrinfo', this.hostname));
+    this.callback(errnoException(err, 'getaddrinfo', this.hostname));
+    return;
   }
 
+  const results = Array(addresses.length);
```

Comment:
For array instantiation like this `new` should be used. Also, `results.push()` below will need to be changed too.